### PR TITLE
feat: apply correct specificity when wrapping a styled component with another

### PR DESCRIPTION
**Pull request by satya164**
Previously, if you had a `Title = styled.h1` and `Heading = styled(Title)`, and then have the same CSS properties specified in both, the styles which are later in the stylesheet would take precedence. This is one of the limitations we have (had) compared to styled-components. But if we can evaluate expressions at build time, we can get the parent className and generate a more specific selector, ensuring that the correct styles are applied. This PR aims to implement that.

### CODE REVIEW

**Comment by satya164** [comment]
I tested on the website and haven't added tests. If someone wants to help by adding tests, much appreciated :)

File: src/babel/extract.js [hunk 1]
```diff
@@ -166,14 +163,14 @@ module.exports = function extract(
           tag.arguments.length === 1 &&
           tag.callee.name === 'styled'
         ) {
-          styled = { component: tag.arguments[0] };
+          styled = { component: path.get('tag').get('arguments')[0] };
```
Comment (thymikee) [inline src/babel/extract.js]: why this change? is `path.get('tag')` different that `tag` from `path.node`?

Comment (satya164) [reply src/babel/extract.js]: `path.node.tag` will return a `node`, but `path.get('tag')` will return a `path`. We need a `path` to be able to traverse it.

**Comment by mmakarin** [comment]
It's happening to me again. Can you double check it?
